node/bser/index.js

```diff
@@ -120,6 +120,9 @@ Accumulator.prototype.peekInt = function(size) {
 
 Accumulator.prototype.readInt = function(bytes) {
   var ival = this.peekInt(bytes);
+  if (ival instanceof Int64 && isFinite(ival.valueOf())) {
+    ival = ival.valueOf();
+  }
   this.readOffset += bytes;
   return ival;
```

Comment:
As I wrote in the other comment, it already returns both depending on the size of the integer.